includes/site-health/audit-autoloaded-options/hooks.php

```diff
@@ -26,3 +26,86 @@ function perflab_aao_add_autoloaded_options_test( $tests ) {
 	return $tests;
 }
 add_filter( 'site_status_tests', 'perflab_aao_add_autoloaded_options_test' );
+
+/**
+ * Register admin actions for handling autoload enable/disable.
+ *
+ * @since n.e.x.t
+ */
+function perflab_aao_register_admin_actions() {
+	add_action( 'admin_action_perflab_aao_update_autoload', 'perflab_aao_handle_update_autoload' );
+}
+add_action( 'admin_init', 'perflab_aao_register_admin_actions' );
+
+/**
+ * Callback for handling disable autoload action.
+ *
+ * @since n.e.x.t
+ */
+function perflab_aao_handle_update_autoload() {
+	check_admin_referer( 'perflab_aao_update_autoload' );
+
+	if ( ! isset( $_GET['option_name'], $_GET['autoload'] ) ) {
+		wp_die( esc_html__( 'Missing required parameter.', 'performance-lab' ) );
+	}
+
+	$option_name = sanitize_text_field( wp_unslash( $_GET['option_name'] ) );
+	$autoload    = isset( $_GET['autoload'] ) ? rest_sanitize_boolean( $_GET['autoload'] ) : false;
+
+	if ( ! current_user_can( 'manage_options' ) ) {
+		wp_die( esc_html__( 'Permission denied.', 'performance-lab' ) );
+	}
+
+	if ( empty( $option_name ) ) {
+		wp_die( esc_html__( 'Invalid option name.', 'performance-lab' ) );
+	}
+
+	// Check if the option exists.
+	if ( false === get_option( $option_name ) ) {
+		// Option doesn't exist, return an error or handle the situation accordingly.
+		wp_die( esc_html__( 'The option does not exist.', 'performance-lab' ) );
+	}
+
+	$result = wp_set_option_autoload( $option_name, $autoload );
+
+	if ( $result ) {
+		// Update disabled options list.
+		$disabled_options = get_option( 'perflab_aao_disabled_options', array() );
+		if ( ! $autoload ) {
+			$disabled_options[] = $option_name;
+		} elseif ( in_array( $option_name, $disabled_options, true ) ) {
+			$key = array_search( $option_name, $disabled_options, true );
+			unset( $disabled_options[ $key ] );
+		}
```

Comment:
Two small improvements can be made here:
* It's worth ensuring that the same option can't be added to the `$disabled_options` list twice.
* You don't need to use both `in_array()` and `array_search()`. If you need the key, you can use `array_search()` alone.

For example:
```suggestion

		$key = array_search( $option_name, $disabled_options, true );
		if ( ! $autoload && false === $key ) {
			$disabled_options[] = $option_name;
		} elseif ( $autoload && false !== $key ) {
			unset( $disabled_options[ $key ] );
		}

```